### DISCUSSION TITLE
Attempted to load class "Asset" from namespace "AppBundle\Controller\Pimcore\Model"

### DISCUSSION THREAD

**Discussion by esisl** [pimcore/pimcore#11311] [Q&A / Support] [answered]
See https://pimcore.com/docs/pimcore/current/Development_Documentation/Assets/Working_with_PHP_API.html
Try:

**<?php
namespace AppBundle\Controller;
...
use Pimcore\Model\Asset;
...
$newAsset = new Pimcore\Model\Asset();**

Get error:
"[2022-01-31 12:23:57] app.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Attempted to load class "Asset" from namespace "AppBundle\Controller\Pimcore\Model". Did you forget a "use" statement for e.g. "Google_Service_CloudAsset_Asset", "Google_Service_Digitalassetlinks_Asset", "Google_Service_DisplayVideo_Asset", "Google_Service_PolyService_Asset", "Google_Service_SecurityCommandCenter_Asset", "Pimcore\Model\Asset", "Pimcore\Model\User\Workspace\Asset", "Pimcore\Model\Asset\MetaData\ClassDefinition\Data\Asset" or "Pimcore\Model\Webservice\Data\Asset"? in /vol/www/sites/medicom/src/AppBundle/Controller/TestController.php:256 Stack trace: #0 /vol/www/pimcore/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(158): AppBundle\Controller\TestController->uploadPublish() #1 /vol/www/pimcore/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(80): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #2 /vol/www/pimcore/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(201): Symfony\Component\HttpKernel\HttpKernel->handle() #3 /vol/www/pimcore/medicom/web/app.php(35): Symfony\Component\HttpKernel\Kernel->handle() #4 {main} [] []"

What's wrong?

**Comment by jdreesen** [comment, marked as answer]
You should either do it like this:

```php
<?php
namespace AppBundle\Controller;
...
use Pimcore\Model\Asset;
...
$newAsset = new Asset();
```

or this:

```php
<?php
namespace AppBundle\Controller;
...
$newAsset = new \Pimcore\Model\Asset();
```

But if you do it like this (note the missing backslash in front of the namespace after `new`), the namespace is wrong, because it will be resolved as `AppBundle\Controller\Pimcore\Model\Asset` (as the error message says).

```php
<?php
namespace AppBundle\Controller;
...
use Pimcore\Model\Asset; // this is not used here
...
$newAsset = new Pimcore\Model\Asset(); // note the missing backslash
```

**Reply by dvesh3** [reply]
Docs fixed here https://github.com/pimcore/pimcore/pull/11313

**Comment by esisl** [comment]
Riddle..

It ONLY works like this:
**<?php
namespace AppBundle\Controller;
...
use Pimcore\Model\Asset; // this is not used here
...
$newAsset = new \Pimcore\Model\Asset();**

**Reply by jdreesen** [reply]
Well, that's fine, then.

**Comment by esisl** [comment]
:D
